### DISCUSSION TITLE
Reason to use `translateY(${virtualRow.start}px)`

### DISCUSSION THREAD

**Discussion by vovsemenv** [TanStack/virtual#828] [Q&A] [answered]
I'm trying to integrate library in the big project and found some problems with click event coordinates using `translateY(${virtualRow.start}px)` and changed it to just `top: `${virtualRow.start}px`.

What is the motivation of using translateY in official docs? browser rendering performance?

**Comment by piecyk** [comment, marked as answer]
Yes, performance reasons. I think you can use getBoundingClientRect to calculate correct position, or use `top` as you mentioned.

**Reply by vovsemenv** [reply]
The problem is that i have to get exact click coordinates to render contextMenu, not item coordinates itself. using `top` for now

<img width="673" alt="image" src="https://github.com/user-attachments/assets/55eaa8e7-a0c9-41ed-9089-ae5bd311e981">

**Comment by vovsemenv** [comment]
Okay also for anyone curios.  FYI @piecyk .
I found one more reason to use `translateY`.
In the safari 17-17.5 tanstack-virtual with `top` is broken because of this bug https://bugs.webkit.org/show_bug.cgi?id=263996
and items in the list flickering. 


https://github.com/user-attachments/assets/74ed0eaf-cfa1-4b4b-8f97-39ead2ad620e
